src/wp_wolfprov.c

```diff
@@ -997,6 +997,10 @@ int wolfssl_provider_init(const OSSL_CORE_HANDLE* handle,
     int ok = 1;
     OSSL_FUNC_core_get_libctx_fn* c_get_libctx = NULL;
 
+#ifdef HAVE_FIPS
+    PRIVATE_KEY_UNLOCK();
```

Comment:
There are unlocks and locks around every private key operation.
Why is this needed and is there a lock call?

Reply:
It's needed for the HKDF extract call. It could be placed just around that function call.

Reply:
Please do.
I don't want global unlocks going on.
Thanks!

Reply:
updated, thanks for the review